GD/__init__.py

```diff
@@ -11,16 +15,203 @@
 from mpv import MPV
 from importlib import reload
 
+logger = logging.getLogger(__name__)
+
+
+class BaseTapeDownloader(abc.ABC):
+    """Abstract base class for a Grateful Dead tape downloader.
+
+    Use one of the subclasses: TapeDownloader or AsyncTapeDownloader.
+    """
+
+    def __init__(self, url="https://archive.org"):
+        self.url = url
+        self.api = f"{self.url}/services/search/v1/scrape"
+        fields = ["identifier", "date", "avg_rating", "num_reviews",
+                  "num_favorites", "stars", "downloads", "files_count",
+                  "format", "collection", "source", "subject", "type"]
+        sorts = ["date asc", "avg_rating desc",
+                 "num_favorites desc", "downloads desc"]
+        self.parms = {'debug': 'false',
+                      'xvar': 'production',
+                      'total_only': 'false',
+                      'count': '10000',
+                      'sorts': ",".join(sorts),
+                      'fields': ",".join(fields)}
+
+    @abc.abstractmethod
+    def get_tapes(self, years):
+        """Get a list of tapes."""
+        pass
+
+
+class TapeDownloader(BaseTapeDownloader):
```

Comment:
I pulled this TapeDownloader code out of GDArchive to make it easy for you to compare and understand the ansync version I wrote with the original.